meta/data.go

```diff
@@ -622,6 +622,11 @@ func (sgi *ShardGroupInfo) Contains(timestamp time.Time) bool {
 	return !sgi.StartTime.After(timestamp) && sgi.EndTime.After(timestamp)
 }
 
+// Overlaps return whether the shard group contains data for the time range between min and max
+func (sgi *ShardGroupInfo) Overlaps(min, max time.Time) bool {
+	return !sgi.StartTime.After(max) && sgi.EndTime.After(min)
+}
```

Comment:
@corylanou in your example, the interval between min and max *should* overlap with 2 shards (as it does).  if you change the min to be `3 * time.Hour` then you only match one shard.